### DISCUSSION TITLE
How can I obtain xpath, the same I can get throught devtools?

### DISCUSSION THREAD

**Discussion by lofti198** [cefsharp/CefSharp#4162] [Q&A] [answered]
How can I programmatically get this xpath:

![image](https://user-images.githubusercontent.com/42960488/178680956-a59ea902-e619-4ccb-9ad2-40cf3ebc8cb4.png)

I know I can use OnFrameLoadEnd to hook onmouseup , where I can add my own custom logic to build xpath based on element clicked and than via CefSharp.PostMessage send it to OnBrowserJavascriptMessageReceived . However is there a way to get that particular xpath I'm getting through devtools?

**Comment by amaitland** [comment, marked as answer]
> However is there a way to get that particular xpath I'm getting through devtools?

You can look at the [DevTools FrontEnd UI](https://github.com/ChromeDevTools/devtools-frontend) source to see exactly how it's done in `Chrome`.
